src/provider/Provider.ts

```diff
@@ -494,4 +492,23 @@ export class Provider extends Instantiable {
       return false
     }
   }
+
+  private async getServiceEndpoint(did: string): Promise<string> {
```

Comment:
Why do we need this? We already have `getServiceEndpoints` that returns all the endpoints. This function seems to return the compute service if found, and if not return the access service. But what happens if you have both? This is kind of hardcoding. A better approach would be to add `serviceIndex` as a function argument and then check that service. The one who uses this function needs to decide what service he wants to use.

Reply:
Yes, we can change to serviceIndex. 

getServiceEndpoints will return the endpoint base on config, there is no issue if we want to check fileinfo by passing url.
If we need to check the fileinfo for existing DID, and the provider from DDO is not the same with endpoint from config (wallet connection), we are checking the wrong endpoint.

Reply:
> getServiceEndpoints will return the endpoint base on config, there is no issue if we want to check fileinfo by passing url.

Yes, my bad.

I'm a little confused about the argument name `url`, isn't that a did? Or can you give me some examples of what url are you refering to?

Reply:
the argument url can be use as:

- file url: 

- mostly we are using for data publish

![image](https://user-images.githubusercontent.com/71874741/114044599-f2836080-98b9-11eb-979c-8c7b8f531c73.png)

- DID:

- provider connectivity check once DDO is crated. in this case we need to get the provider from DDO instead for config.

![image](https://user-images.githubusercontent.com/71874741/114044698-03cc6d00-98ba-11eb-98e6-4ee0848f9a5a.png)

Reply:
I'm not sure is it better to split this function into 2

- filelinfo: pass url as argument and check the connectivity base on endpoint from config and return File response
- isFileconsumable: pass did as argument and check the connectivity base on endpoint from DDO and return Boolean

Reply:
sounds good, less confusing and more usability

Reply:
I'm going to create isFileConsumable then.
Is it better have 2 argument: did and serviceIndex?